src/ml/uq.hpp

```diff
@@ -0,0 +1,98 @@
+/*
+ * Copyright 2021-2023 Lawrence Livermore National Security, LLC and other
+ * AMSLib Project Developers
+ *
+ * SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+ */
+
+#ifndef __AMS_UQ_HPP__
+#define __AMS_UQ_HPP__
+
+#include <stdexcept>
+#include <vector>
+
+#include "AMS.h"
+#include "ml/hdcache.hpp"
+#include "ml/surrogate.hpp"
+#include "wf/resource_manager.hpp"
+
+template <typename FPTypeValue>
+class UQ
+{
+public:
+PERFFASPECT()
+  static void evaluate(
+      AMSUQPolicy uqPolicy,
+      const int totalElements,
+      std::vector<const FPTypeValue *> &inputs,
+      std::vector<FPTypeValue *> &outputs,
+      const std::shared_ptr<HDCache<FPTypeValue>> &hdcache,
+      const std::shared_ptr<SurrogateModel<FPTypeValue>> &surrogate,
+      bool *p_ml_acceptable)
+  {
+    if ((uqPolicy == AMSUQPolicy::DeltaUQ_Mean) ||
+        (uqPolicy == AMSUQPolicy::DeltaUQ_Max)) {
+      CALIPER(CALI_MARK_BEGIN("DELTAUQ");)
+      const size_t ndims = outputs.size();
+      std::vector<FPTypeValue *> outputs_stdev(ndims);
+      // TODO: Enable device-side allocation and predicate calculation.
+      for (int dim = 0; dim < ndims; ++dim)
+        outputs_stdev[dim] =
+            ams::ResourceManager::allocate<FPTypeValue>(totalElements,
+                                                        AMSResourceType::HOST);
+
+      CALIPER(CALI_MARK_BEGIN("SURROGATE");)
+      DBG(Workflow, "Model exists, I am calling DeltaUQ surrogate (for all data)");
```

Comment:
I don't like the identifier is Workflow. Since we have not finished the design of the new interface leave it as is.

Reply:
agreed